File: 2025/zig/src/template.zig
const std = @import("std");
const adlib = @import("adlib.zig");

fn partOne(input: []const u8) u32 {
    _ = input;
    return 0;
}

fn partTwo(input: []const u8) u32 {
    _ = input;
    return 0;
}

pub fn main() !void {
    const alloc = adlib.allocator;
    const input = try adlib.collectStdin(alloc);
    const res_1 = partOne(input);
    const res_2 = partTwo(input);
    std.debug.print("part one: {d}\npart two {d}\n", .{ res_1, res_2 });
}

File: 2025/ts/src/template.ts
import * as adlib from "./adlib.ts";

function partOne(input: string): number {
    return 0;
}

function partTwo(input: string): number {
    return 0;
}

export async function main() {
    const input = await new Response(Deno.stdin.readable).text();

    const res = partOne(input);
    const res2 = partTwo(input);

    console.log(`part one: ${res}\npart two: ${res2}\n`);
}

if (import.meta.main) await main();
